src/webots/nodes/utils/WbNodeUtilities.cpp

```diff
@@ -663,7 +663,24 @@ WbMatter *WbNodeUtilities::findUppermostMatter(WbNode *node) {
 }
 
 WbSolid *WbNodeUtilities::findTopSolid(const WbNode *node) {
-  return dynamic_cast<WbSolid *>(const_cast<WbNode *>(findTopNode(node)));
+  if (node == NULL || node->isWorldRoot())
```

Comment:
Testing `node->isWorldRoot()` might not be necessary, as long as a node which get `node->isWorldRoot() == true` also provide `node->parentNode() == NULL` (which must be confirmed, I'm not certain on that one, event if it sound logical). In that case, your code handle nicely this case, and return `NULL` as well.